### ISSUE TITLE
Pick<any, string | number> is not any

### ISSUE THREAD

**Issue by acrazing** [Microsoft/TypeScript#26320]
code example:

```ts
interface Foo<T> {
  data: T;
}

declare function foo<T extends Foo<{ object: string; }>>(input: T): void

declare const a: Foo<Pick<any, string | symbol | number>>;

foo(a);
```

Emit error:

```bash
error TS2345: Argument of type 'Foo<Pick<any, string | number | symbol>>' is not assignable to parameter of type 'Foo<{ object: string; }>'.
  Type 'Pick<any, string | number | symbol>' is not assignable to type '{ object: string; }'.
    Property 'object' is missing in type 'Pick<any, string | number | symbol>'.
```

Expected behavior: no error

typescript version: `3.0.1`

playground: <https://www.typescriptlang.org/play/#src=%0D%0Ainterface%20Foo%3CT%3E%20%7B%0D%0A%20%20data%3A%20T%3B%0D%0A%7D%0D%0A%0D%0Adeclare%20function%20foo%3CT%20extends%20Foo%3C%7B%20object%3A%20string%3B%20%7D%3E%3E(input%3A%20T)%3A%20void%0D%0A%0D%0Adeclare%20const%20a%3A%20Foo%3CPick%3Cany%2C%20string%20%7C%20symbol%20%7C%20number%3E%3E%3B%0D%0A%0D%0Afoo(a)%3B%0D%0A>

**Comment by RyanCavanaugh** [comment]
`Pick` only copies over the specified keys from `T`, so it forms a type of known keys that doesn't need to be `any`.
